### ISSUE TITLE
[EPIC] Check and review permissions

### ISSUE THREAD

**Issue by Nesqwik** [lenra-io/lenra_cli#95]
Good practices should be defined. For example :  
- Add `@derive {Jason.Encoder, only: [...]}` for every ecto schema to ensure that potentially critical new added field are not encoded and returned to the API (unless specifically allowed)
- Add a base rule that deny every API request to any Route if not specifically allowed 
- Check if there is no better way to check permission than Bouncer. (using a plug maybe ?)
   - https://github.com/cpjk/canary
   - https://github.com/boydm/policy_wonk
   - https://github.com/schrockwell/bodyguard

A bunch of check and test should be done to ensure that the permission/access are correctly setup.
- Is good practices are respected everywhere ?
- Check that we cannot access API/Data route if we are not connected. (Add unit test if needed)
- Check that we cannot access the Data from the "Data API" from another app/environment. (Add unit test if needed)
- More generally, review every route to check : 
  - Can we ask for data that we should not be able to see ?
  - Can we store/update/delete data that we should not be able to write ?
  - Is the returned data contain sensible information (credential, hashed password, private data...)
  - Can we find any security flaw ?
  - Is there is unit test that check everything ?

Suggestion : 
We could add a test directory in every project that need it to group all the security unit test. Using naming convention, we can fast check if a route is fully covered from unit test or not.

**Comment by taorepoara** [comment]
@Nesqwik Is it managed by #209 ?

**Comment by Nesqwik** [comment]
No, this only check that the token have the correct scope. Not the permissions.
This one relates to Bouncer, we must make sure that no resource is available to the wrong person and no-one can update/delete anything that he should not be able to.
